src/commands.rs

```diff
@@ -0,0 +1,212 @@
+#[derive(PartialEq, Debug)]
```

Comment:
Perhaps if this was a module it would help namespace all of these types, so their names would not have to contain "Command" and from outside it would be clearer that they are a part of one module.

Reply:
Since the module itself is called `commands.rs`, I think it's sufficient just to remove `Command` and access it from other modules as `commands::Begin` in stead of `BeginCommand`.